test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -1773,6 +1790,27 @@ static int create_tm_system(void)
 	dynamic_threshold_update = tm_capabilities.dynamic_threshold_update;
 	dynamic_wred_update = tm_capabilities.dynamic_wred_update;
 
+	tm_shaper_min_rate = tm_capabilities.per_level[0].min_rate;
+	tm_shaper_max_rate = tm_capabilities.per_level[0].max_rate;
+	tm_shaper_min_burst = tm_capabilities.per_level[0].min_burst;
+	tm_shaper_max_burst = tm_capabilities.per_level[0].max_burst;
+	for (level = 1; level < NUM_LEVELS; level++) {
+		odp_tm_level_capabilities_t *per_level =
+			&tm_capabilities.per_level[level];
+
+		if (per_level->min_rate > tm_shaper_min_rate)
+			tm_shaper_min_rate = per_level->min_rate;
+
+		if (per_level->min_burst > tm_shaper_min_burst)
+			tm_shaper_min_burst = per_level->min_burst;
+
+		if (per_level->max_rate < tm_shaper_max_rate)
+			tm_shaper_max_rate = per_level->max_rate;
+
+		if (per_level->max_burst < tm_shaper_max_burst)
+			tm_shaper_max_burst = per_level->max_burst;
+	}
```

Comment:
If different levels have non-overlapping allowed ranges of the parameters, after this loop it could be that min_rate > max_rate. I suggest checking for that and giving up the test in that case.

Reply:
Ack.